index.js

```diff
@@ -1,18 +1,15 @@
 const markdownToHtml = require('./lib/markdown-to-html');
 const getPageData = require('./lib/get-page-data');
+const configData = require('./lib/config-data');
 const renderHtml = require('./lib/render-html');
-const getHeadings = require('./lib/get-headings');
 
-function formatter(markdown) {
+function formatter(markdown, options) {
 
     return markdownToHtml(markdown)
         .then(html => getPageData(html))
-        .then(data => addHeadings(data))
+        .then(data => configData(data, options))
         .then(data => renderHtml(data))
-}
-
-function addHeadings(data) {
-    return Object.assign({}, data, { headings: getHeadings(data.body) });
+        .catch(err => { throw err });
```

Comment:
This code does the same if you leave out this line. You catch an error and rethrow it. Other modules consuming `formatter` could just chain on their `.catch` without this line. So leave it out.